File: java/app/src/main/java/net/fbvictorhugo/j/barreirasanitaria/data/dao/dummy/UsuarioDummyDataBase.java
package net.fbvictorhugo.j.barreirasanitaria.data.dao.dummy;

import net.fbvictorhugo.j.barreirasanitaria.data.dao.IUsuarioDAO;
import net.fbvictorhugo.j.barreirasanitaria.data.model.Usuario;

import java.util.Date;

public class UsuarioDummyDataBase implements IUsuarioDAO {

    private static UsuarioDummyDataBase sFactory;

    public static UsuarioDummyDataBase getInstance() {
        if (sFactory == null) {
            sFactory = new UsuarioDummyDataBase();
        }
        return sFactory;
    }

    private UsuarioDummyDataBase() {
    }

    @Override
    public Usuario procurarUsuario(String login, String senha) {
        if (login != null && !login.isEmpty() && senha != null && !senha.isEmpty()) {
            Usuario usuario = new Usuario();
            usuario.setId(1);
            usuario.setLogin(login);
            usuario.setSenha(senha);
            usuario.setNome("Victor Hugo");
            usuario.setUltimoLogin(new Date());
            return usuario;
        } else {
            return null;
        }
    }

}

File: kotlin/app/src/main/java/net/fbvictorhugo/k/barreirasanitaria/data/dao/dummy/UsuarioDummyDataBase.kt
package net.fbvictorhugo.k.barreirasanitaria.data.dao.dummy

import net.fbvictorhugo.k.barreirasanitaria.data.dao.IUsuarioDAO
import net.fbvictorhugo.k.barreirasanitaria.data.model.Usuario

object UsuarioDummyDataBase : IUsuarioDAO {

    private val mDummyList: ArrayList<Usuario> = ArrayList()

    init {
        val usuario = Usuario("Victor Hugo")
        usuario.apply {
            id = 1
            login = "abc"
            senha = "123"
        }
        mDummyList.add(usuario)
    }

    override fun procurarUsuario(login: String, senha: String): Usuario? {
        if (login.trim().isNotEmpty() && senha.trim().isNotEmpty()) {
            for (usuario in mDummyList) {
                if (usuario.login == login && usuario.senha == senha) {
                    return usuario
                }
            }
        }
        return null
    }

}
